### PR TITLE
Update Gold Member marketing

### PR DESCRIPTION
**Pull request by davidfischer**
## Changes

In general, the goal of this PR is to update some of our marketing and documentation to better reflect the reality at Read the Docs.

Here's an overview of the changes:

- Make it more obvious that becoming a Gold Member removes advertising
- Try to standardize on the terminology "Gold Member" rather than "buying gold" or anything else.
- Explain about RTD advertising in the user profile where users can go community-only, donate, or otherwise opt-out/-in
- When `settings.USE_PROMOS = False`, remove the links to manage advertising.

One change I did not do but am considering is to allow staying logged-in longer than 2 weeks. This becomes somewhat important when users are becoming a Gold Member to skip ads. Having people log in every two weeks seems suboptimal.

### CODE REVIEW

**Comment by davidfischer** [comment]
I don't think this lint error is related to these changes...

File: docs/ethical-advertising.rst [hunk 1]
```diff
@@ -135,6 +135,9 @@ Opting Out
 
 We have added multiple ways to opt out of the advertising on Read the Docs.
 
+Users can go ad-free for as long as they are logged-in
+by becoming a `Gold Member to Read the Docs <https://readthedocs.org/accounts/gold/>`_.
```
Comment (ericholscher) [inline docs/ethical-advertising.rst]: For some reason the `to` feels weird here. I guess it implies a subscription? That makes sense, just reads weird to me and I'm not 100% sure why.

File: docs/support.rst [hunk 2]
```diff
@@ -17,10 +17,9 @@ Good questions for Stack Overflow would be:
 Community Support
 -----------------
 
-Read the Docs is a community supported site,
-nobody is paid to handle readthedocs.org support.
-We are hoping to bring in enough money with our `Gold`_ program to change that,
-so please sign up if you are able.
+Read the Docs is a community- and ad-supported site.
```
Comment (ericholscher) [inline docs/support.rst]: `community-` is a bit odd here. I think maybe reword it:

Read the Docs is supported by community contributions and advertising.

File: readthedocs/gold/templates/gold/subscription_form.html [hunk 3]
```diff
@@ -59,6 +59,9 @@ <h2>Read the Docs Gold</h2>
       we suggest giving at least $20/month to help cover our support and operations costs.
```
Comment (ericholscher) [inline readthedocs/gold/templates/gold/subscription_form.html]: I feel like these other parts of the page are just blabbing in an awkward way also. I'm almost tempted to just remove them?

File: readthedocs/profiles/views.py [hunk 4]
```diff
@@ -279,3 +280,34 @@ def profile_detail(
 
     context.update({'profile': profile_obj})
     return render(request, template_name, context=context)
+
+
+def account_advertising(request):
```
Comment (ericholscher) [inline readthedocs/profiles/views.py]: Does this need a login_requires decorator? Guessing it will blow up with a non-logged in user otherwise?

File: docs/ethical-advertising.rst (hunk 1 above)
Comment (davidfischer) [reply docs/ethical-advertising.rst]: I think `of` is better than `to`.

File: readthedocs/gold/templates/gold/subscription_form.html (hunk 3 above)
Comment (davidfischer) [reply readthedocs/gold/templates/gold/subscription_form.html]: I'll give it a read and see if there's anything that should be removed. I tried to remove anything inaccurate already.

Comment (davidfischer) [reply readthedocs/gold/templates/gold/subscription_form.html]: I do think everything around [domains](https://github.com/davidfischer/readthedocs.org/blob/a466bf6385ffbd5f758c9f56b0fe9840816ae089/readthedocs/gold/templates/gold/subscription_form.html#L71-L90) can be deleted once we get Lets Encrypt support. I don't quite think I'll delete that yet.

Comment (ericholscher) [reply readthedocs/gold/templates/gold/subscription_form.html]: Why would LE support change our infra cost for domains? If anything, it would increase the overhead we have for each domain we serve, managing the certs, etc.

**Comment by davidfischer** [comment]
> Why would LE support change our infra cost for domains? 

It would be fully automated at that point. There would be no marginal cost to additional domains once the work on LE is done. Currently there's a manual step there.

**Comment by ericholscher** [comment]
What manual step? We don't support SSL at all for them currently. The idea was for people using CNAME's to pay more, because they were getting more value and we were getting less brand awareness.

**Comment by davidfischer** [comment]
Ahhh. I thought we did support it if they gave us the cert. I guess not.
